### ISSUE TITLE
Should we collect courses like what OSSU is doing

### ISSUE THREAD

**Issue by nhanvu327** [mithi/robotics-coursework#17]
Hi,

I'm really interested in robotics lately. Would love if we plan to implement something like this https://github.com/ossu/computer-science but for robotics.

**Comment by mithi** [comment]
> Hi,
> 
> I'm really interested in robotics lately. Would love if we plan to implement something like this https://github.com/ossu/computer-science but for robotics.

Hey, this is really cool, but I'm not part of any active online robotics communities online such as discord servers...
are you active in any robotics discord servers ?

**Comment by nhanvu327** [comment]
I just follow a robotics thread in Reddit. https://www.reddit.com/r/robotics/

**Comment by mithi** [comment]
> I just follow a robotics page in Reddit. https://www.reddit.com/r/robotics/

Okay cool. If you mean to make a repo so that the free robotics courses are structured like a formal school (like OSSU did),
I'm all for it. And if you mean to create an active online discord community to hold people accountable for finishing the curriculum the way it is structured, that's great too.

**Comment by nhanvu327** [comment]
> > I just follow a robotics page in Reddit. https://www.reddit.com/r/robotics/
> 
> Okay cool. If you mean to make a repo so that the free robotics courses are structured like a formal school (like OSSU did),
> I'm all for it. And if you mean to create an active online discord community to hold people accountable for finishing the curriculum the way it is structured, that's great too.

Yeah, I'm gonna find a solid syllabus and find courses. How is the convenient way to contact you ?

**Comment by mithi** [comment]
> Yeah, I'm gonna find a solid syllabus and find courses. How is the convenient way to contact you ?

You can contact me here in this issue 😀

**Comment by nhanvu327** [comment]
> > Yeah, I'm gonna find a solid syllabus and find courses. How is the convenient way to contact you ?
> 
> You can contact me here in this issue 😀

Cool, sure. About lab manuals, I am taking the self-learning path and I see the most important aspect that a self-learning lacks compared to uni learner is lab experience / hands-on experience. I think we need more detailed instructions, equipment necessary for practice.

**Comment by mithi** [comment]
> Cool, sure. About lab manuals, I am taking the self-learning path and I see the most important aspect that a self-learning lacks compared to uni learner is lab experience / hands-on experience. I think we need more detailed instructions, equipment necessary for practice.

Okay.

I've shared on the other closed issue: 

A lot of courses listed in this repo have great lab exercises, for example:
- https://www.edx.org/course/self-driving-cars-with-duckietown
    - http://duckietown.com/
- https://alliance.seas.upenn.edu/~meam620/wiki/index.php?n=Main.Projects
- https://www.autonomousrobotslab.com/student-projects2.html
- http://prg.cs.umd.edu/enae788m

**Comment by hot9cups** [comment]
Hello @mithi 

Firstly, this repo is great, thank you for putting it together.

I was actually looking for exactly what this issue is for - 'OSSU but for robotics' - and that's how I stumbled upon this.

I see a fair bit of content on the repo, but I'm really unsure what to pick and what order to go. I also noticed you were a robotics teacher in the past.
Do you have suggestions/recommendations on how one should navigate through this repo, and with robotics courses in general?
Where to begin, how to know what to do next, and what to aim for in the long term?

Would appreciate any and all help!

**Comment by mithi** [comment]
Hi @hot9cups , 

I think it depends on how you want to go? Do you want to go the Maker route or the Academic route?  There are also many aspects of robotics, it's a fusion of mechanical, electrical engineering and computer science. What aspect are you most interested in? Also what's your motivation? Is it a serious hobby, do you want to pursue a professional career in robotics, or are you just testing the waters to see if it's right for you?


 In any case, I advice the following:

# Maker Route
1.  If you're starting from zero and have zero knowledge about making, you can start familiarizing yourself with arduino, raspberry pi and different kinds of motors:
  - https://github.com/mithi/robotics-coursework/blob/master/PROTOTYPING.md
2. Then if you still don't feel that confident, you can try whatever interests you in Adafruit:
  - https://learn.adafruit.com/search?q=robotics 
3. After getting comfortable, you can build more advanced stuff like:     

 -  your own self driving car : https://docs.donkeycar.com/    
  - quadcopter from scratch http://www.brokking.net/ymfc-32_main.html 
  - and/or a quadruped https://github.com/PetoiCamp/OpenCat , https://www.petoi.com/

4. ??? 
5. Profit!!  (build anything you like 😁)

# Academic Route 

Assuming you have basic familiarity in algebra and coding:  

1. Start with:  https://robotacademy.net.au/ (free, no login required)

2. After that, basic fundamental knowledge in the following are essential 
    - Math fundamentals for robotics: https://github.com/interesting-courses/16-811, https://www.cs.cmu.edu/~me/811/
    - Sensor Fusion (Kalman Filters) https://github.com/balzer82/Kalman 
    - Control systems https://github.com/calcmogul/controls-engineering-in-frc 
    - Classical robotics algorithms https://github.com/AtsushiSakai/PythonRobotics
    - Basic Machine Learning https://www.cs.cornell.edu/courses/cs4780/2024sp/

3.  You can go anywhere from there depending on where your interest lies, but I believe the most lucrative one is the Artificial Intelligence Route - and it's moving quite fast and quite hard to keep up. Start reading papers in Deep Learning for robotics (Reinforcement Learning, Computer Vision), here's a few links to help you get started, they curated a lot of readings 
    - CS 6758: Deep Learning for Robotics (Fall 2024) Cornell University https://www.cs.cornell.edu/courses/cs6758/2024fa/
    - CMU Deep Learning for Robotics 16-884, Fall 2022 https://16-884.github.io/lectures/
    - Advanced Deep Learning for Robotics ADLR CIT433027 https://bbaeuml.github.io/tum-adlr/adlr/lectures.html

----

Hope that helps and hope you enjoy your robotics learning adventure 😁

**Comment by Hope-land321** [comment]
Thanks a lot for the courses. I was looking for a comprehensive and full roadmap to help me pursue my dream of becoming a robotics engineer, and I came across Mr. Lukas M. Ziegler's LinkedIn post.
